server/src/main/java/io/deephaven/server/console/ConsoleServiceGrpcImpl.java

```diff
@@ -185,12 +186,13 @@ public void executeCommand(ExecuteCommandRequest request, StreamObserver<Execute
     public void getHeapInfo(GetHeapInfoRequest request, StreamObserver<GetHeapInfoResponse> responseObserver) {
         GrpcUtil.rpcWrapper(log, responseObserver, () -> {
             final RuntimeMemory runtimeMemory = RuntimeMemory.getInstance();
+            final Sample sample = new Sample();
+            runtimeMemory.readOnly(sample);
```

Comment:
I think you're right - this was unintentional, and the getHeapInfo should call `read`, triggering an update (`if now >= snapshot.nextCheck`).

On Enterprise this class is different, but checking `totalMemory()` calls a `maybeUpdateValues()` (and Enterprise side could also benefit from reading it all into one sample, as it's susceptible to the volatile read as well - @cpwright for comment).

Side note I don't know why we need to pass in our own buff - why not just get the current snapshot? The entire snapshot gets replaced on each read. The `reset()` and `copy()` function are never called on `Sample`.

Reply:
The runtime memory doesn't give you data atomically for free/total anyway so I'm not very concerned about the millisecond caching maybe giving you a cached value for one but fetching for the next.

Reply:
I'm not *too* concerned about getting values from inconsistent Snapshots (since the underlying calls to Runtime are independent anyways); I would like to use the "best API possible" though, and `Sample` is the way to do that IMO. (I'm going to motivate in a future patch to add timestamp b/c I think that's an obvious omission.)

Reply:
Snapshot is a private class, that's why we can't just return it. I think we could definitely improve RuntimeMemory in this regards, but this was meant to be a simple PR - I'd prefer follow-ups PRs if necessary.

Reply:
I've updated the PR to trigger snapshot when necessary from getHeapInfo